client/finality-grandpa/src/lib.rs

```diff
@@ -541,6 +541,22 @@ where
 	)
 }
 
+/// A descriptor for an authority set hard fork. These are authority set changes
+/// that are not signalled by the runtime and instead are defined off-chain
+/// (hence the hard fork).
+pub struct AuthoritySetHardFork<Block: BlockT> {
+	/// The new authority set id.
+	pub set_id: SetId,
+	/// The block hash and number at which the hard fork should be applied.
+	pub block: (Block::Hash, NumberFor<Block>),
+	/// The authorities in the new set.
+	pub authorities: AuthorityList,
+	/// The last finalized block number by the network. If defined, then the
+	/// authority set change will be forced, i.e. the node won't wait for the
+	/// block above to be finalized before enacting it.
```

Comment:
This comment confuses me.

I mean I get what it does by reading the code, but I interpret the comment differently.

Reply:
I updated it. Let me know if it is more understandable now.